### ISSUE TITLE
lots of small tracks

### ISSUE THREAD

**Issue by wrohdewald** [tinuzz/wp-plugin-trackserver#22]
It has been a while since I checked my track list in your plugin. Version 4.0.2 is installed.

Now I suddenly have about 18000 separate tracks uploaded by  Oruxmaps v7.2.14. A new track is automaticallly started every few seconds with about 2-20 points per track.

This is too much for merging via web interface, so I want to extract all tracks from the data base, in whatever form - I can then merge that and convert to gpx if needed.

How would I get the tracks out of the data base? (of course I could read the source code, but I hope you can save me some time) I do not use mysql too often, so a little help would be appreciated. Like what is the database name, where can I find login credentials, which tables should I look at?

How can this happen? If it is not Oruxmaps, could it be that this huge amount of tracks slows down your server so much that it runs into some timeout, resetting and clearing the current lifetrack connection? I would have to check when this started to happen - but the web interface is currently as good as unusable. Would have to check in the data base.

Next on my todo-list: Since I do not want to do anything in php, I will write my own mapmytracks server in python which will work as a proxy, forwarding data to your server.  That will allow me to easier find out what exactly happens and perhaps merge future data within my proxy. Maybe even write data directly into your data base.

**Comment by tinuzz** [comment]
Hi,

I am sorry for not responding to your issue report. I was on holiday when you wrote it, and I completely forgot about it afterwards. I haven't looked at Trackserver or done any development for quite a while, mostly due to lack of time (or priority, some might say).

Did you manage to solve the issue?

I have no idea how it might have happened. I too sometimes have some tracks from OruxMaps in my database with 0 or only a few locations, and I haven't really found the cause for those, but it's never more than a few, and certainly not _thousands_!

To answer some of your questions anyway (even though I assume I'm way too late): the Trackserver database is just your WordPress database, so you should know what it is called and what the login credentials are. Trackserver keeps two tables for itself: {prefix}ts_tracks and {prefix}ts_locations, where {prefix} is your WordPress database table name prefix, which defaults to 'wp_'.

Merging tracks is nothing more than giving all the involved locations the same track ID (or trip_id, as the table column is called), so if you know the trip_id of one of the locations and you know the IDs of the first and last location you want to merge into one track, you could do this:

`UPDATE wp_ts_locations SET trip_id=$trip_id WHERE id >= $first_point_id AND id <= $last_point_id`

where you would substitute $trip_id, $first_point_id and $last_point_id appropriately.

**Comment by tinuzz** [comment]
Oh and I forgot to comment on the slowness: like you, I have noticed that WordPress as a whole becomes slower if there are many tracks and locations in the Trackserver database. My database periodically builds up to a few hundred tracks with something like 1.5 million locations, and my whole site becomes noticeably slower when that happens.

I have not looked into the cause for that, but I suspect there must be database queries happening that aren't directly related to publishing or managing data from Trackserver, maybe metadata queries by WordPress itself, or something similar. I'll have to look into that to find out what's going on.

**Comment by wrohdewald** [comment]
I did find some bugs with Oruxmaps: https://www.oruxmaps.com/foro/viewtopic.php?f=5&t=5244
But even then it remains strange. Meanwhile I have Python code which can act as a Mapmytracks server, and it can directly update your data base. But right now I am trying something else:

I am writing a small Android app which acts as a Mapmytracks client. It does not determine GPS positions itself but whenever any other app does, my app will use that point and send it out. When that works (the sending part is missing), I can use both clients (Oruxmaps and mine) in parallel to send to my server and hopefully figure out more by comparing the server logs. When the transfer from smartphone works correctly and there are still problems, I will look into the receiving side on your end.

Is there some way for an external program directly writing into your database to tell your plugin to refresh the track display? That way I would not have to use the POST interface.

**Comment by tinuzz** [comment]
I see you're investing quite some time in this ;-) I have also found strange behaviour in OruxMaps from time to time, but I feel that trying to get them fixed is futile. As long as OruxMaps is not open source, I'm not going to try anymore.

In Trackserver, writing to the database (either via one of the GET or POST interfaces, or directly) and displaying tracks are completely unrelated things. When live tracking is enabled (which you can force by setting `live=true` in the shortcode), the display just refreshes the track every 10 seconds. It's just a JavaScript setInterval() construct, that gets the track data and re-renders it on the map.

This is done client-side, so there is no trigger you can pull on the server side. You could modify the code to do more frequent updates. 10 seconds is hard coded at this time.

https://github.com/tinuzz/wp-plugin-trackserver/blob/master/trackserver.js#L532

Add 'interval=<milliseconds>' to the options hash of the L.control.liveupdate call.

https://github.com/tinuzz/wp-plugin-trackserver/blob/master/lib/leaflet-liveupdate-1.1/leaflet-liveupdate.js

Cheers,
Martijn.

**Comment by tinuzz** [comment]
By the way,writing a simple, but stable MapMyTracks client for Android was on my own TODO list for some time. I like TrackMe a lot, it's stable and has only a little it of feature bloat,, but it's not open source either, and I don't really like its protocol. MapMyTracks is far better, but the only usable client is OruxMaps, which isn't all that great.

So... would you make your Android code open source? Would you be interested in developing and sharing it as a usable Trackserver client?

**Comment by tinuzz** [comment]
In late 2019, I made some improvements to the MapMyTracks handling code in Trackserver. I once observed the problem with many small tracks myself. This happens when a new track is created in Trackserver, but saving the initial trackpoints fails, leading to an error in OruxMaps. On the next request, OruxMaps re-sends the 'start_actvity' request, leading to the creation of a second track, but again failing to save the locations. And so on.

I think I fixed this in the mean time. The fix will be part of the next release  If it ever happens again, feel free to re-open this.
